[Woo POS] 12386 Fix for Cart animated overlay not showing

samiuelson opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: #12386 
<!-- Id number of the GitHub issue this PR addresses. -->

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
This PR fixes the animated overlay in the Cart pane which stopped showing up after recent code changes. 

The overlay with animated opacity was wrongly placed in the layout hierarchy.

### Steps to reproduce
<!-- Step-by-step testing instructions. For new user flows, consider instead stating the goal of the workflow and see if your PR reviewer can accomplish the workflow without specific steps! -->
Observe that Cart pane is not being dimmed out when cart is empty.

### Testing information
<!-- This is your opportunity to break out individual scenarios that need testing (when necessary) and/or include a checklist for the reviewer to go through. Consider documenting the following from your own completed testing: devices used, alternate workflows, edge cases, affected areas, critical flows, areas not tested, and any remaining unknowns. Provide feedback on this new section of the PR template through Sept 30, 2024 to Apps Quality; additional context here: p91TBi-b8z-p2#comment-12036 -->
The cart pane should be dimmed when it's empty, and active when there is at least one item in the cart.

### The tests that have been performed
<!-- To give the reviewer an idea of what could be missed in terms of testing -->
I tested adding and removing items from a cart in both light and dark modes.

### Images/gif
<!-- Include before and after images or gifs when appropriate. -->
https://github.com/user-attachments/assets/dff2537f-e732-4f43-950d-78ab664fc15f

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [ ] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [ ] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [ ] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

samiuelson committed:
Update screen structure to make overlay visible

samiuelson committed:
Extract overlay to standalone fun

dangermattic commented:
<!--
  0 Errors
  1 Warning: This PR is assigned to the mil...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/232">20.1</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

samiuelson committed:
Align Products and Cart toolbar sizes

samiuelson committed:
Change Card to Surface

Card contains Surface internally; always one-level less in the layout hierarchy

samiuelson committed:
Revert "Change Card to Surface"

This reverts commit 63f30ff80b6ef036b6719011c4d207aa685b4624.

samiuelson committed:
Revert "Align Products and Cart toolbar sizes"

This reverts commit 556312a79dd16fb6925fdf3b28134207da3a6c81.

backwardstruck committed:
Merge branch 'trunk' into 12386-cart-overlay-fix

backwardstruck committed:
Merge branch 'trunk' into 12386-cart-overlay-fix

backwardstruck commented:
Tested in dark and light mode:

![Screenshot_20240822_112751](https://github.com/user-attachments/assets/0edc2275-5c35-4dbb-a9a3-6ad4022a2b54)
![Screenshot_20240822_112716](https://github.com/user-attachments/assets/4d0ee3d9-41d1-4a7c-ad2f-13d47271e777)

backwardstruck reviewed (approved):
Tested and code change looks correct.
